### DISCUSSION TITLE
Is it possible for the 3ds to support flock?

### DISCUSSION THREAD

**Discussion by kaylambdadelta** [rust3ds/ctru-rs#200] [Q&A] [answered]
I am using the wonderful `redb` crate and it harmlessly assumes all unix family os's outside of wasi have flock support, which prevents use on the 3ds (missing pieces in libc), is it possible for this to be implemented for the 3ds or what?

**Comment by FenrirWolf** [comment, marked as answer]
The 3DS has no notion of file locking, so I doubt that the `flock` API could be properly supported. That being said, I took a look at `redb`'s source code and it looks like it has a [fallback API](https://github.com/cberner/redb/blob/master/src/tree_store/page_store/file_backend/fallback.rs) that just uses mutexes for file locking. So to me it looks like the best way to add 3DS support to `redb` would be to make a PR to [this file](https://github.com/cberner/redb/blob/master/src/tree_store/page_store/file_backend/mod.rs) that chooses the fallback impl for the `horizon` OS instead of using the `unix` impl.

**Comment by kaylambdadelta** [comment]
I see, thank you!
